### ISSUE TITLE
It doesn't run on Debian 11 - where is the source code

### ISSUE THREAD

**Issue by osowiecki** [NovembreLab/admixture#1]
Where is the source code and where is the information about your build process. What system was used to build the binary package?

**Comment by Alex2PO** [comment]
Still no fix?

**Comment by vatanparast** [comment]
I confirm that in Debian 12.11 both of 64 and 32 bit versions fail, either by `Segmentation fault` or `bad_alloc`. However, no issue in redhat or rocky linux. I assume the binaries are compiled at rpm based.

Please release the source code!

**Comment by dalexander** [comment]
Hi there.  I've built a new ADMIXTURE binary, and I'd appreciate if you could test it out?

Please see https://dalexander.github.io/admixture/candidate.html and let me know if this new binary works for you.

Regarding open-sourcing-- ADMIXTURE is not open source at this time but we may consider opening it in the future.

**Comment by vatanparast** [comment]
Hello, thank you for the update. The new version worked as expected in Debian 12, kernel 6.1.147-1 (2025-08-02) x86_64.

**Comment by dalexander** [comment]
If you have a chance, I'd love to hear the binary remains compatible with your older systems as well.  Thanks!

**Comment by joansaurina** [comment]
Dear ADMIXTURE authors,

First of all, thank you for developing such a foundational tool for the community.

We are currently working on our own implementation and would like to ask for some clarification regarding how exactly the cross-validation (CV) procedure is performed internally in ADMIXTURE.

Since the original source code isn't public, it is still a bit unclear to us from the original paper how the CV training loop operates. 

Specifically, we have two main technical questions:

1. Parameter Initialization: For each of the v folds, do you re-train the P and Q parameters entirely from scratch (e.g., random initialization), or do you use the global estimates (or estimates from previous folds) as a warm start to speed up convergence?

2. Optimization Algorithm: Does the optimization during the CV step use the exact same block-relaxation / quasi-Newton EM updates as the main algorithm, or is a simpler EM approach used for these intermediate steps?

We were wondering if you could perhaps share a small pseudocode snippet or a brief explanation of how this CV procedure is implemented. 

This would be incredibly helpful to ensure our implementation faithfully matches the original method.

Thank you for your time and help!

Best regards,

Joan Saurinai-i-Ricos, Alexander G. Ioannidis

**Comment by dalexander** [comment]
Joan and Alexander-- to your questions.  (Some of this may overlap with content in the manual or in the paper [Enhancements to the ADMIXTURE algorithm for individual ancestry estimation](https://link.springer.com/article/10.1186/1471-2105-12-246) but I do think there are some missing details in those.

  1. Initialization: warm-started from the global fit.

  For each of the v folds, P and Q are initialized from the global estimates produced by the main run on the full dataset — not from random, and not chained off the previous fold. Every fold starts from the same warm-start point.

  2. Algorithm: identical to the main run, with a looser stopping rule.

  The per-fold optimization uses the same quasi-Newton-accelerated block-relaxation updates as the main algorithm — same factory, same acceleration, same penalty. The only difference is the convergence criterion: the main run iterates to a objective change tolerance (default 1e-4), while each CV fold runs only a small fixed number of polishing iterations (default 3)
  from the warm start. Both can be overridden on the command line.

  3. Folding scheme: per-entry masking.

 A "fold" is a random subset of genotype entries (not individuals or loci) marked as missing. The optimizer already handles missing data, so each fold is simply a re-fit on a different missingness pattern over the same I×L matrix. The CV score on each fold is the sum of squared deviance residuals between the true genotype at each held-out entry and its imputation 2·Qᵢᵀ Pₗ from the refit; the final CV index is that sum divided by the number of non-missing entries.

  Pseudocode:

```
  fit main algorithm to full data  ->  global Q, P  (Δ < 1e-4)

  assign each genotype entry a random fold label in {1..v}
  cvSum = 0
  for fold in 1..v:
      mask all entries with this fold's label as MISSING
      initialize Q_cv, P_cv from the global Q, P     # warm start
      run the same QN-accelerated block-relaxation updates,
          stopping after a small fixed number of iterations (default 3)
      cvSum += squared deviance residuals on the held-out entries
  cvIndex = cvSum / (# non-missing entries)
```

 So in short: warm-started from the global fit, same QN-accelerated block-relaxation updates as the main algorithm, only a few polishing iterations per fold, masking at the genotype-entry level.
